gpbft/participant.go

```diff
@@ -151,6 +151,49 @@ func (p *Participant) ReceiveAlarm() (err error) {
 	return nil
 }
 
+// FinalityInfo is an auxiliary struct to pass relevant info from the finality certificate
+// NOTE: It is not clear to me what information is available in the finality certificate
+// and in what form. The information provided here may change, using it as a placeholder for now.
+type FinalityInfo struct {
+	Instance uint64
+	Power    *PowerTable
+	Beacon   []byte
+}
+
+func (p *Participant) ReceiveFinalityCertificate(f FinalityInfo) error {
+	// Get ready to start the instance defined in the finality certificate for the next instance
+	p.gpbft = nil
+	prevInstance := p.currentInstance
+	// Set the instance from the certificate as the current one.
+	p.currentInstance = f.Instance
+	// clean all messages queued for instances below the one in the finality certificate
+	// and old committees
+	for i := prevInstance + 1; i < p.currentInstance; i++ {
+		delete(p.mqueue.messages, i)
+		delete(p.committees, i)
+	}
```

Comment:
I suggest adding a `nextInstance` parameter to `finishCurrentInstance()` and then calling that method. Update it to clear the queues in the same way you have here.